azurelinuxagent/common/dhcp.py

```diff
@@ -80,23 +79,30 @@ def wireserver_route_exists(self):
         This is true when running in a virtual network.
         :return: True if a route to KNOWN_WIRESERVER_IP exists.
         """
-        route_exists=False
+        route_exists = False
         logger.info("test for route to {0}".format(KNOWN_WIRESERVER_IP))
         try:
-            route = shellutil.run_get_output("grep -c {0} /proc/net/route".format(KNOWN_WIRESERVER_IP_ENTRY))
+            # grep -c returns 1 if the count is 0, so ignore the return code
+            route = shellutil.run_get_output(
```

Comment:
The feedback in #232 seems to be based on resiliency, while that above on performance. Since this check is executed once on a relatively small file, I'm not sure performance is enough of a concern to warrant a change (note that this is not code being introduced with this PR). With regards to resiliency, I agree with the feedback that parsing the content of a file should be done with Python itself, but for a simple check like this I am not sure I agree.

Let me know if you folks feel strongly on this one, and we can chat more.